ports/minimal/Makefile

```diff
@@ -19,12 +19,12 @@ INC += -I$(BUILD)
 ifeq ($(CROSS), 1)
 DFU = $(TOP)/tools/dfu.py
 PYDFU = $(TOP)/tools/pydfu.py
-CFLAGS_CORTEX_M4 = -mthumb -mtune=cortex-m4 -mcpu=cortex-m4 -msoft-float -fsingle-precision-constant -Wdouble-promotion
+CFLAGS_CORTEX_M4 = -mthumb -mtune=cortex-m4 -mcpu=cortex-m4 -msoft-float -fsingle-precision-constant -Wdouble-promotion -Wfloat-conversion
 CFLAGS = $(INC) -Wall -Werror -std=c99 -nostdlib $(CFLAGS_CORTEX_M4) $(COPT)
 LDFLAGS = -nostdlib -T stm32f405.ld -Map=$@.map --cref --gc-sections
 else
 LD = gcc
-CFLAGS = -m32 $(INC) -Wall -Werror -std=c99 $(COPT)
+CFLAGS = -m32 $(INC) -Wall -Wdouble-promotion -Werror -std=c99 $(COPT)
```

Comment:
Does it make sense to enable float warnings on ports like this that don't enable floats at all?

Reply:
Not sure, if not enabling flotas means there effectively isn't a single line of code using float or double it doesn't matter of course. But you could also say it doesn't hurt, and it's there already in case someone uses this as a base for another port or just plays around and does enable floats.

Btw I think I just forgot -Wfloat-conversion here.

But in any case: in general do you think this is the way to go, having as much ports as possible use it (some of the warnings found were only because of that) and each makefile specifying it's own flags?

Reply:
Also: should the commits enabling the warnings all be squashed, and do we put them before or after the ones with the conversion fixes?

Reply:
Definitely it should be up to each port to decide and enable the warnings that it wants to enable (ie these settings shouldn't be centralised).

I guess it makes sense (and doesn't hurt) to have these enabled in those ports that could enable floats.  As you say, as a basis for another port.

> Also: should the commits enabling the warnings all be squashed, and do we put them before or after the ones with the conversion fixes?

Yeah, squashing them seems sensible (so one commit that adds all warnings to all ports/variants).  And putting them after the fixes would be my preference, so that each commit still builds without error (generally try to do this, eg add tests after fixing a bug, but not always possible).